cibuildwheel/macos.py

```diff
@@ -93,236 +78,191 @@ def get_python_configurations(
     return [c for c in python_configurations if build_selector(c.identifier)]
 
 
-SYMLINKS_DIR = Path("/tmp/cibw_bin")
-
-
-def make_symlinks(installation_bin_path: Path, python_executable: str, pip_executable: str) -> None:
-    assert (installation_bin_path / python_executable).exists()
-
-    # Python bin folders on Mac don't symlink `python3` to `python`, and neither
-    # does PyPy for `pypy` or `pypy3`, so we do that so `python` and `pip` always
-    # point to the active configuration.
-    if SYMLINKS_DIR.exists():
-        shutil.rmtree(SYMLINKS_DIR)
-    SYMLINKS_DIR.mkdir(parents=True)
-
-    (SYMLINKS_DIR / "python").symlink_to(installation_bin_path / python_executable)
-    (SYMLINKS_DIR / "python-config").symlink_to(
-        installation_bin_path / (python_executable + "-config")
-    )
-    (SYMLINKS_DIR / "pip").symlink_to(installation_bin_path / pip_executable)
-
-
 def install_cpython(version: str, url: str) -> Path:
-    installed_system_packages = subprocess.run(
-        ["pkgutil", "--pkgs"], universal_newlines=True, check=True, stdout=subprocess.PIPE
-    ).stdout.splitlines()
+    installed_system_packages = call("pkgutil", "--pkgs", text=True).splitlines()
 
     # if this version of python isn't installed, get it from python.org and install
     python_package_identifier = f"org.python.Python.PythonFramework-{version}"
-    python_executable = "python3"
-    installation_bin_path = Path(f"/Library/Frameworks/Python.framework/Versions/{version}/bin")
+    installation_path = Path(f"/Library/Frameworks/Python.framework/Versions/{version}")
```

Comment:
This is still global, right?

Reply:
Ahh, it gets cleaned afterwards if it wasn't pre-installed? And if it was pre-installed it's an error? (this will probably be easier to read directly instead of as a diff, so I'll try to do that tomorrow)

Reply:
I will edit the description for macOS as it's a bit more complex (for local runs).
It's still global, not checked for an exact match and never cleaned up.
All installs are never cleaned up (same on macOS and Windows). On Windows and for macOS PyPy, we're sure we'll be using the exact version specified by cibuildwheel. Not the case for python.org installers on macOS (same as before in fact).

Reply:
This now looks like:
```
Building cp36-macosx_x86_64 wheel
CPython 3.6 macOS x86_64

Installing Python cp36...

+ pkgutil --pkgs
Error: CPython 3.6 is not installed.
cibuildwheel will not perform system-wide installs when running outside of CI.
To build locally, install CPython 3.6 on this machine, or, disable this version of Python using CIBW_SKIP=cp36-macosx_*

Download link: https://www.python.org/ftp/python/3.6.8/python-3.6.8-macosx10.9.pkg
```